### PR TITLE
feat: ability to upgrade the cluster via the new binary

### PR DESCRIPTION
**Pull request by emosbaugh**

### CODE REVIEW

File: .github/workflows/release-dev.yaml [hunk 1]
```diff
@@ -30,15 +30,15 @@ jobs:
 
       - name: Build and push local-artifact-mirror image
         run: |
-          export SHORT_SHA=dev-$(git rev-parse --short=7 HEAD)
+          export SHORT_SHA=dev-${GITHUB_SHA::7}
```
Comment (laverya) [inline .github/workflows/release-dev.yaml:33]: this should match what's in the pull-request.yaml version of things

**Review by laverya — COMMENTED** [review]
Looks good besides the SHORT_SHA mismatch between pull-request.yaml and release-dev.yaml (we should really combine those soon)

**Review by sgalsaleh — APPROVED** [review]

File: .github/workflows/release-dev.yaml (hunk 1 above)
Comment (sgalsaleh) [reply .github/workflows/release-dev.yaml:33]: can't use the same thing here though as `github.event.pull_request.head.sha` doesn't apply here, no?

Comment (sgalsaleh) [reply .github/workflows/release-dev.yaml:33]: i guess we could use `GITHUB_SHA` in `pull-request.yaml`?

**Review by laverya — APPROVED** [review]

**Pull request merged**